### ISSUE TITLE
Kernel panic on EC2 reboot

### ISSUE THREAD

**Issue by eqyiel** [NixOS/nixops#410]
On unstable channel.  I'm not sure how to debug this myself.  I can create an instance just fine and connect to it, but if I reboot it will become completely unreachable and need to be terminated.

Relevant bits from the system log:
```
[    0.000000] Cannot get hvm parameter CONSOLE_EVTCHN (18): -22!
[    0.250075] Cannot get hvm parameter CONSOLE_EVTCHN (18): -22!

[1;32m<<< NixOS Stage 1 >>>[0m

loading module xen-blkfront...
loading module xen-netfront...
loading module fuse...
loading module dm_mod...
loading module af_packet...
running udev...
bringing up network interface eth0...
bringing up network interface lo...
acquiring IP address via DHCP...
udhcpc (v1.23.2) started
Sending discover...
Sending select for 172.31.21.0...
Lease of 172.31.21.0 obtained, lease time 3600
networking is up!
starting device mapper and LVM...
NOCHANGE: partition 1 is size 41940992. it cannot be grown
[    0.880011] Kernel panic - not syncing: Attempted to kill init! exitcode=0x00000100
[    0.880011]
[    0.880011] CPU: 0 PID: 1 Comm: init Not tainted 4.4.2 #1-NixOS
[    0.880011] Hardware name: Xen HVM domU, BIOS 4.2.amazon 12/07/2015
[    0.880011]  ffff88003e2bbeb0 ffff88003e2bbe20 ffffffff812b3700 ffffffff817181c0
[    0.880011]  ffff88003e2bbea0 ffffffff81148839 ffffffff00000010 ffff88003e2bbeb0
[    0.880011]  ffff88003e2bbe48 ffffffff81141bc1 0000000000000100 ffff88003e308010
[    0.880011] Call Trace:
[    0.880011]  [<ffffffff812b3700>] dump_stack+0x44/0x64
[    0.880011]  [<ffffffff81148839>] panic+0xc6/0x203
[    0.880011]  [<ffffffff81141bc1>] ? perf_event_exit_task+0xc1/0x370
[    0.880011]  [<ffffffff8107067c>] do_exit+0xa8c/0xad0
[    0.880011]  [<ffffffff810709bf>] ? SyS_wait4+0x6f/0xc0
[    0.880011]  [<ffffffff81070745>] do_group_exit+0x45/0xb0
[    0.880011]  [<ffffffff810707c4>] SyS_exit_group+0x14/0x20
[    0.880011]  [<ffffffff814e08ae>] entry_SYSCALL_64_fastpath+0x12/0x71
[    0.880011] Kernel Offset: disabled
```

And it'll hang there forever.

My ec2-specific configuration is pretty standard, just bits and pieces copied from the Nixops manual:

```
let
  region = "ap-southeast-2";
  # accessKeyId = "digital"; # symbolic name looked up in ~/.ec2-keys
in
{
  mangoagogo =
  { config, pkgs, resources, ... }:
  { deployment.targetEnv = "ec2";
    deployment.ec2.accessKeyId = "digital";
    deployment.owners = [ "team@mangochutney.com.au" ];
    deployment.ec2.region = region;
    deployment.ec2.ebsBoot = true;
    deployment.ec2.instanceType = "t2.micro";
    deployment.ec2.securityGroups = [ "default" ];
    deployment.ec2.keyPair = resources.ec2KeyPairs.mango;
    deployment.ec2.elasticIPv4 = "52.63.103.137";
  };

  network.description = "mangoclients.com.au";
  resources.ec2KeyPairs.mango = {
    inherit region;
  };
}
```

Please lend me your insight!

**Comment by kevincox** [comment]
I'm getting the same error on all of my ec2 instances.

**Comment by rbvermaa** [comment]
@kevincox Are you using nixos-unstable as well?

**Comment by kevincox** [comment]
I think I am on nixpkgs-unstable. I don't know the different TBH.

**Comment by rbvermaa** [comment]
I couldn't reproduce on nixos-15.09, but I can reproduce this on nixos-unstable channel. Will try to debug this in the morning.

**Comment by eqyiel** [comment]
@rbvermaa Thanks for looking into it!

**Comment by rbvermaa** [comment]
I have pushed a fix for this in NixOS/nixpkgs@ed5920ec6552be6a1e03b15a21db9f3c792898eb
